engine/api/workflow/hook.go

```diff
@@ -119,7 +117,7 @@ func deleteHookConfiguration(db gorp.SqlExecutor, store cache.Store, p *sdk.Proj
 	}
 	code, errHooks := services.DoJSONRequest(srvs, http.MethodDelete, fmt.Sprintf("/task/bulk"), hookToDelete, nil)
 	if errHooks != nil || code >= 400 {
-		log.Warning("HookRegistration> Unable to delete old hooks [%d]: %s", code, errHooks)
+		return fmt.Errorf("HookRegistration> Unable to delete old hooks [%d]: %s", code, errHooks)
```

Comment:
if you return an error, transaction will be rollbacked => hook will in database be not anymore on gitlab/bitbucket/github.